File: lang/go/src/fun/vac/data_structures/interfaces/IList/IList.go
package IList

type IList interface {

    Size() int

    IsEmpty() bool

    Get(index int) int

    Set(index int, element int)

    Insert(index int, element int)

    Remove(index int)

    Front() int

    Back() int

    Prepend() int

    Append() int

    Poll()

    Eject()
}

File: lang/php/src/vac.fun/data_structures/interfaces/IList.php
<?php
    namespace vac\fun\data_structures\interfaces\IList;

    interface IList {
        public function size(): int;

        public function isEmpty(): bool;

        public function get(int $index): int;

        public function set(int $index, int $element): void;

        public function insert(int $index, int $element): void;

        public function remove(int $index): void;

        public function front(): int;

        public function back(): int;

        public function prepend(int $element): void;

        public function append(int $element): void;

        public function poll(): void;

        public function eject(): void;
    }
?>
